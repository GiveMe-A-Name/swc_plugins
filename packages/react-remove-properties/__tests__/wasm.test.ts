import { expect, test } from "vitest";
import { transform } from "@swc/core";
import path from "node:path";
import url from "node:url";

const pluginName = "swc_plugin_react_remove_properties.wasm";

const transformCode = async (code: string, options = {}) => {
  return transform(code, {
    jsc: {
      parser: {
        syntax: "ecmascript",
        jsx: true,
      },
      target: "es2018",
      experimental: {
        plugins: [
          [
            path.join(
              path.dirname(url.fileURLToPath(import.meta.url)),
              "..",
              pluginName,
            ),
            options,
          ],
        ],
      },
    },
    filename: "test.jsx",
  });
};

test("Should load react-remove-properties wasm plugin correctly", async () => {
  const input = `export default function Home() {
  return (
    <div data-test-id="1" data-custom="1a">
      <div data-custom="2">
        <h1 data-testid="3">Hello World!</h1>
      </div>
    </div>
  );
}
`;
  const { code } = await transformCode(input);
  expect(code).toMatchSnapshot();
});
