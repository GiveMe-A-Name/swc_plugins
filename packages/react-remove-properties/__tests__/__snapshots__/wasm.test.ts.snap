// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Should load react-remove-properties wasm plugin correctly 1`] = `
"export default function Home() {
    return /*#__PURE__*/ React.createElement("div", {
        "data-custom": "1a"
    }, /*#__PURE__*/ React.createElement("div", {
        "data-custom": "2"
    }, /*#__PURE__*/ React.createElement("h1", null, "Hello World!")));
}
"
`;
